src/Leela.cpp

```diff
@@ -265,6 +266,15 @@ int main (int argc, char *argv[]) {
     // Initialize network
     Network::initialize();
 
+    // Initialize cache if playouts is set
+    if (cfg_max_playouts) {
+        // cache hits are generally from last several moves so setting cache
+        // size based on playouts increases the hit rate while balancing memory
+        // usage for low playout instances. 100'000 cache entries is ~500 MB
+        auto max_size = std::min(100'000, std::max(6'000, 3*cfg_max_playouts));
+        NNCache::get_NNCache()->resize(max_size);
+    }
+
```

Comment:
I prefer making a function inside NNCache that takes cfg_max_playouts as the parameter and calculates an appropriate size for itself.

Reply:
That is, the NNCache specific logic for figuring out the size should be in NNCache.cpp, not in Leela.cpp.

Reply:
Updated, I also moved all of the global object initialization (threadpool, rng, zobrist, nncache) into a method.